src/Form/Switch.js

```diff
@@ -1,120 +1,116 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import styled, { css } from 'styled-components';
+import { css } from 'styled-components';
+import { createComponent } from '../utils';
 import { createEasyInput } from './EasyInput';
 
-const SwitchContain = styled.label`
-  position: relative;
-  display: inline-block;
-  width: ${p => p.size * 2}px;
-  height: ${p => p.size + p.inset}px;
-`;
-
-const SwitchInput = styled.input`
-  display: none;
-`;
-
-const SwitchThumb = styled.span`
-  ${({
-    variant,
-    size,
-    inset,
-    on,
-    theme,
-    backgroundColor = theme.colors[variant] || theme.colors[theme.variants[variant]],
-  }) => css`
-    border-radius: ${size * 2}px;
+const SwitchThumb = createComponent({
+  name: 'SwitchThumb',
+  as: 'i',
+  style: ({ trackInset, thumbSize, value }) => css`
+    border-radius: ${thumbSize * 2}px;
     position: absolute;
-    cursor: pointer;
     top: 0;
-    left: 0;
-    right: 0;
     bottom: 0;
-    transition: 0.3s;
-    background-color: ${on ? backgroundColor : 'transparent'};
-    border: 2px solid ${on ? backgroundColor : theme.colors.greyLight};
-
-    &:active::before {
-      width: ${size + inset}px;
-
-      ${on &&
-        css`
-          transform: translateX(${size - inset * 2}px);
-        `}
+    left: 0;
+    height: ${thumbSize - trackInset * 2}px;
+    width: ${thumbSize - trackInset * 2}px;
+    background-color: white;
+    box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.1), 0px 4px 4px rgba(0, 0, 0, 0.1);
+    transition: all 0.3s cubic-bezier(0.125, 0.85, 0.3, 1.125);
+
+    ${value &&
+      css`
+        transform: translateX(${thumbSize}px);
+      `};
+  `,
+});
+
+const SwitchTrack = createComponent({
+  name: 'SwitchTrack',
+  as: 'label',
+  style: ({ theme, trackColor, thumbSize, trackInset, value }) => css`
+    position: relative;
+    display: inline-block;
+    cursor: pointer;
+    width: ${thumbSize * 2}px;
+    height: ${thumbSize}px;
+    background: ${value ? theme.colors[trackColor] : 'transparent'};
+    border: ${trackInset}px solid ${value ? theme.colors[trackColor] : theme.colors.grey};
+    border-radius: ${thumbSize * 2}px;
```

Comment:
Maybe add a little fade to those colors?

```suggestion
    border-radius: ${thumbSize * 2}px;
    transition: background-color 0.3s, border-color 0.3s;
```

Reply:
Nice